pkg/chain/ethereum/beacon.go

```diff
@@ -169,3 +169,22 @@ func (bc *BeaconChain) UpdateOperatorStatus() error {
 	_, err := bc.randomBeacon.UpdateOperatorStatus(bc.key.Address)
 	return err
 }
+
+// IsEligibleForRewards checks whether the operator is eligible for rewards or not.
+func (bc *BeaconChain) IsEligibleForRewards() (bool, error) {
+	// TODO: check if bc.key.Id.ID() is the right argument
```

Comment:
No, this is not the right argument. The ID expected by the sortition pool is defined [here](https://github.com/keep-network/sortition-pools/blob/b3fcb43e09d9fc2304fa2f65635aec406a6cfe79/contracts/SortitionTree.sol#L50) and `bc.key.Id` is - based on my understanding - a random UUID to simplify lookups in the keystore.

I've just opened the PR to sortition-pools to make the API of rewards eligibility consistent and always use the operator address: https://github.com/keep-network/sortition-pools/pull/182

Keep in mind that once #182 is merged, we will need to regenerate contract bindings. We can do it in this PR.

Reply:
Right, that's why I put a TODO to double-check this, it didn't seem right to me but I couldn't find the correct ID that I'd be 100% sure.